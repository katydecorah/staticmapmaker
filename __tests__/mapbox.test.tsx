import { render } from "@testing-library/react";
import Mapbox from "@/pages/mapbox";

it("renders homepage unchanged", () => {
  const { container } = render(<Mapbox />);
  expect(container).toMatchSnapshot();
});
