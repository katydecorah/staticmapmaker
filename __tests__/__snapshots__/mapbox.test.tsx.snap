// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="col-left"
    >
      <div
        class="sticky"
      >
        <header
          class="header"
        >
          <div
            class="title"
          >
            <h1>
              <a
                href="https://staticmapmaker.com"
              >
                <svg
                  aria-hidden="true"
                  class="header-icon"
                  data-icon="map"
                  focusable="false"
                  height="128"
                  role="img"
                  viewBox="0 0 128 128"
                  width="128"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.5 0C2.367 0 0 2.367 0 5.5v117c0 3.133 2.367 5.5 5.5 5.5h117c3.133 0 5.5-2.367 5.5-5.5V5.5c0-3.133-2.367-5.5-5.5-5.5H5.5zM8 8h112v112H8V8zm35.188 2l15 15-9.594 9.594a2 2 0 1 0 2.813 2.813L61 27.813 101.19 68l-50.032 50h5.688l47.156-47.187 14 14v-5.625L106.814 68l11.188-11.188v-5.625l-14 14L63.814 25l15-15H73.19L61 22.186 48.814 10h-5.625zM48 52c-8.8 0-16 7.2-16 16 0 12 16 32 16 32s16-20 16-32c0-8.8-7.2-16-16-16zm0 8c4.4 0 8 3.6 8 8s-3.6 8-8 8-8-3.6-8-8 3.6-8 8-8z"
                  />
                </svg>
                <div>
                  Mapbox
                   Static Map Maker
                </div>
              </a>
            </h1>
          </div>
        </header>
        <div
          class="controls"
        >
          <div
            class="container"
          >
            <div
              class="fieldset error fieldset-row"
            >
              <label
                class="label"
                for="api"
              >
                API Token
              </label>
              <input
                id="api"
                placeholder="Public API Key (required)"
                type="text"
                value=""
              />
            </div>
            <div
              class="fieldset  fieldset-row"
            >
              <label
                class="label"
                for="mapid"
              >
                Your style
              </label>
              <input
                id="mapid"
                placeholder="username/style_id"
                type="text"
                value=""
              />
            </div>
            <div
              class="fieldset  fieldset-row"
            >
              <label
                class="label"
                for="mapboxID"
              >
                Mapbox style
              </label>
              <select
                id="mapboxID"
              >
                <option
                  value="mapbox/streets-v11"
                >
                  Streets
                </option>
                <option
                  value="mapbox/light-v10"
                >
                  Light
                </option>
                <option
                  value="mapbox/dark-v10"
                >
                  Dark
                </option>
                <option
                  value="mapbox/satellite-v9"
                >
                  Satellite
                </option>
                <option
                  value="mapbox/satellite-streets-v11"
                >
                  Satellite streets
                </option>
                <option
                  value="mapbox/outdoors-v11"
                >
                  Outdoors
                </option>
              </select>
            </div>
            <div
              class="fieldset  fieldset-row"
            >
              <label
                class="label"
                for="width"
              >
                Width
              </label>
              <input
                id="width"
                max="1280"
                min="0"
                type="number"
                value="600"
              />
            </div>
            <div
              class="fieldset  fieldset-row"
            >
              <label
                class="label"
                for="height"
              >
                Height
              </label>
              <input
                id="height"
                max="1280"
                min="0"
                type="number"
                value="300"
              />
            </div>
            <div
              class="fieldset checkbox-fieldset fieldsetRow"
            >
              <input
                checked=""
                id="retina"
                type="checkbox"
              />
              <label
                class="checkbox-label"
                for="retina"
              >
                Retina maps
              </label>
            </div>
            <div
              class="fieldset checkbox-fieldset fieldsetRow"
            >
              <input
                checked=""
                id="attribution"
                type="checkbox"
              />
              <label
                class="checkbox-label"
                for="attribution"
              >
                Show attribution
              </label>
            </div>
            <div
              class="fieldset checkbox-fieldset fieldsetRow"
            >
              <input
                checked=""
                id="logo"
                type="checkbox"
              />
              <label
                class="checkbox-label"
                for="logo"
              >
                Show logo
              </label>
            </div>
            <div
              class="fieldset  fieldset-row"
            >
              <label
                class="label"
                for="location"
              >
                Long, lat
              </label>
              <input
                id="location"
                type="text"
                value="-73.7638,42.6564"
              />
            </div>
            <div
              class="fieldset  fieldset-row"
            >
              <label
                class="label"
                for="zoom"
              >
                Zoom
              </label>
              <input
                id="zoom"
                max="22"
                min="0"
                step="0.25"
                type="range"
                value="13"
              />
            </div>
            <div
              class="fieldset  fieldset-row"
            >
              <label
                class="label"
                for="bearing"
              >
                Bearing (rotate)
              </label>
              <input
                id="bearing"
                max="360"
                min="0"
                type="range"
                value="0"
              />
            </div>
            <div
              class="fieldset  fieldset-row"
            >
              <label
                class="label"
                for="pitch"
              >
                Pitch (tilt)
              </label>
              <input
                id="pitch"
                max="60"
                min="0"
                type="range"
                value="0"
              />
            </div>
            <div
              class="fieldset checkbox-fieldset fieldsetRow"
            >
              <input
                id="auto"
                type="checkbox"
              />
              <label
                class="checkbox-label"
                for="auto"
              >
                Fit map to markers
              </label>
            </div>
            <div
              class="fieldset"
            >
              <div
                class="form-group"
              >
                <button
                  class="btn"
                >
                  <svg
                    class="feather"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <line
                      x1="12"
                      x2="12"
                      y1="5"
                      y2="19"
                    />
                    <line
                      x1="5"
                      x2="19"
                      y1="12"
                      y2="12"
                    />
                  </svg>
                   Add a marker
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-right"
    >
      <div
        class="static"
      >
        <div
          class="static-wrapper api-required"
        >
          <img
            alt="Mapbox Map of -73.7638,42.6564"
            class="static-map"
            id="map"
            src="/img/mapbox-placeholder.png"
            width="600"
          />
        </div>
        <div
          class="attribution"
        >
          © 
          <a
            href="https://www.mapbox.com/about/maps/"
          >
            Mapbox
          </a>
           ©
           
          <a
            href="http://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           
          <strong>
            <a
              href="https://www.mapbox.com/map-feedback/"
              rel="noreferrer"
              target="_blank"
            >
              Improve this map
            </a>
          </strong>
        </div>
      </div>
      <div
        class="wrapper"
      >
        <div
          class="info"
        >
          <h2>
            Request
          </h2>
          <div
            class="code"
          >
            <pre>
              https://api.mapbox.com/styles/v1/mapbox/outdoors-v11/static/-73.7638,42.6564,13,0,0/600x300@2x?attribution=true&logo=true&access_token=YOUR-ACCESS-TOKEN-HERE
            </pre>
          </div>
          Display the map as an image:
          <div
            class="code"
          >
            <pre>
              &lt;img width="
              600
              " src="
              https://api.mapbox.com/styles/v1/mapbox/outdoors-v11/static/-73.7638,42.6564,13,0,0/600x300@2x?attribution=true&logo=true&access_token=YOUR-ACCESS-TOKEN-HERE
              " alt="
              Mapbox map of -73.7638,42.6564
              " /&gt;
            </pre>
          </div>
          Display the map as a CSS background image:
          <div
            class="code"
          >
            <pre>
              background-image: url(
              https://api.mapbox.com/styles/v1/mapbox/outdoors-v11/static/-73.7638,42.6564,13,0,0/600x300@2x?attribution=true&logo=true&access_token=YOUR-ACCESS-TOKEN-HERE
              );
            </pre>
          </div>
          <h2>
            How to use
          </h2>
          <ol>
            <li>
              Get a 
              <a
                href="https://docs.mapbox.com/api/overview/#access-tokens-and-token-scopes"
              >
                Mapbox
                 API key
              </a>
              .
            </li>
            <li>
              Customize the fields to build your map.
            </li>
            <li>
              Copy and paste the generated code into your HTML document or a stylesheet.
            </li>
          </ol>
          <h2>
            About
          </h2>
          <p>
            You can add a static map to any website with simple HTML and no JavaScript. Using URL parameters, you can describe the type of map that you want and 
            Mapbox
             will send it back.
          </p>
          <p>
            The Static Map Maker helps you configure the 
            Mapbox
             API parameters.
          </p>
          <p>
            Check out 
            <a
              href="https://docs.mapbox.com/api/maps/#static-images"
            >
              Mapbox
              's static maps API
            </a>
            .
          </p>
        </div>
      </div>
    </div>
  </main>
</div>
`;
